src/storage/datasets_index.rs

```diff
@@ -13,18 +13,19 @@ use super::layout::DataChunk;
 #[derive(Default)]
 pub struct DatasetsIndex {
     datasets: HashMap<Arc<Dataset>, DatasetIndex>,
+    pub chunks_ordinals_map: HashMap<DataChunk, u64>,
     http_headers: reqwest::header::HeaderMap,
 }
 
 #[derive(Debug, PartialEq, Eq)]
 pub struct RemoteFile {
     pub url: Url,
-    pub name: Arc<str>,
+    pub name: String,
 }
 
 struct DatasetIndex {
     url: Url,
-    files: HashMap<DataChunk, Vec<Arc<str>>>,
+    files: HashMap<DataChunk, Vec<String>>,
```

Comment:
Why is it only using chunk IDs without datasets as a key? What if different datasets have chunks with the same ids?

Reply:
done